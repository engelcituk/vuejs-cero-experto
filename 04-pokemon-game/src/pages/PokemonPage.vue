<template>
    <div>
        <h1>¿Quién es este pokemón?</h1>
        <PokemonPicture :pokemonId="251" :showPokemon="true" />
        <PokemonOptions :pokemons="pokemonArr" />
    </div>
</template>

<script>
    import PokemonPicture from '@/components/PokemonPicture'
    import PokemonOptions from '@/components/PokemonOptions'
    import getPokemonOptions from '@/helpers/getPokemonOptions'

    getPokemonOptions()
    export default {
        components:{
            PokemonPicture,
            PokemonOptions
        },
        data(){
            return {
                pokemonArr: []
            }
        },
        methods:{
            async mixPokemonArr(){
                this.pokemonArr = await getPokemonOptions()
                console.log(this.pokemonArr)
            }
        },
        mounted(){
            this.mixPokemonArr()
        }
    }
</script>

<style lang="scss" scoped>

</style>