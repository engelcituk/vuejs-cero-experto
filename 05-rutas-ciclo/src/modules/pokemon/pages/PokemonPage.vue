<template>
    <h1>Pokemon Page <span>#{{id}}</span></h1>
</template>

<script>
export default {
    props:{
        id:{
            type: Number,
            required: true
        },
        
    },
    created(){
        // const { id } = this.$route.params
        // this.id = id
        // console.log(this.$attrs)
    } ,
    data(){
        return {
            // id : null
        }
    }  
}
</script>
