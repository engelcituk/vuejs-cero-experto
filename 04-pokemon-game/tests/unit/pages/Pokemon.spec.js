
import { shallowMount, mount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'

describe('Pokemon Page component', ()=> {
    let wrapper
    let mixPokemonArrSpy 

    beforeEach( () => {
        mixPokemonArrSpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArr')
        wrapper = shallowMount(PokemonPage) //monto el PokemonPage, en cada test, asi si se modifica algo en este se vuelve a su estado inicial
    })

    test('Debe de hacer match con el snapshot', ()=> {
        expect(wrapper.html()).toMatchSnapshot() // toMatchInlineSnapshot
    })

    test('Debe de llamar a mixPokemonArr al montar',  () => {
        const mixPokemonArrSpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArr')
        expect(mixPokemonArrSpy).toHaveBeenCalled() 

    })
  
})