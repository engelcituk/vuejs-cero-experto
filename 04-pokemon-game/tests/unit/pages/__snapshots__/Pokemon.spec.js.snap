// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Page component Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>Espere por favor</h1>
  <!--v-if-->
</div>
`;
